### PR TITLE
Gracefully handle invalid CSV files

### PR DESCRIPTION
**Pull request by steveoh**
# Thanks for contributing

## Identify the issue referencing the feature or bug

#117 

## Description of the Change

Catch the errors and show specific instructions for resolving the problem.

### CODE REVIEW

File: src/services/csv.js [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
-      }
+        resolve(data.length);
+      });
+    } catch (parseError) {
+      throw new Error(`${CSV_PARSE_ERROR}: {${parseError.code}} {${parseError.message}}`);
+    }
 
-      resolve(data.length);
-    });
     rs.pipe(parser);
```
Comment (stdavis) [inline src/services/csv.js:36]: Should this be the line that is try/caught? Rather than above where the parser is just created?

**Comment by stdavis** [comment]
This LGTM but I want an approval from @steveoh on my last commit. Then I'm happy to rebase and merge.

**Review by stdavis — APPROVED** [review]

**Comment by stdavis** [comment]
> will you fix the commit message
I was thinking that it should just be fixed up but thought that you would want to look at my changes first.

File: src/services/csv.js [hunk 2]
```diff
@@ -1,29 +1,36 @@
...
   return new Promise((resolve, reject) => {
-    const rs = fs.createReadStream(filePath);
-    const parser = parse({ columns: true }, function (err, data) {
-      if (err) {
-        reject(err);
-
-        return;
-      }
+    const parser = parse({ columns: true }, function (parseError, data) {
+      reject(`${CSV_PARSE_ERROR}: {${parseError.code}} {${parseError.message}}`);
```
Comment (steveoh) [inline src/services/csv.js:24]: **issue**: is this function correct? It looks like it will always call reject and never get to resolve.

**Review by steveoh — COMMENTED** [review]
let's review this together when you get a chance.

**Review by stdavis — APPROVED** [review]

**Pull request merged**